pkg/controller/gitopsconfig/controller.go

```diff
@@ -432,29 +432,47 @@ func (r *Reconciler) manageDeletion(instance *gitopsv1alpha1.GitOpsConfig) (reco
 		return reconcile.Result{}, xerrors.Errorf("GitOpsConfig finalizer unable to list owned jobs for %q: %w", instance.Name, err)
 	}
 
-	// If exactly 1 job exists, but it's blocked because of bad image, we
-	// assume the GitOpsConfig never managed to successfully deploy, so we can
-	// just delete the job, remove the finalizer, and be done (#216). It may be
-	// either action=create or action=delete job.
-	// If a job is blocked because of bad image, it only has one active pod
-	if len(jobs) == 1 && jobs[0].Status.Succeeded == 0 && jobs[0].Status.Failed == 0 && jobs[0].Status.Active == 1 {
-		status, err := jobContainerStatus(context.TODO(), r.client, &jobs[0])
-		if err != nil {
-			log.Error(err, "GitOpsConfig finalizer unable to get job pod's status", "instance", instance.Name)
-			return reconcile.Result{}, xerrors.Errorf("GitOpsConfig finalizer unable to get job pod's status for %q: %w", instance.Name, err)
-		}
-		log.Info("GitOpsConfig finalizer found one job", "instance", instance.Name, "podStatus", status)
-		safeReasons := []string{"ErrImagePull", "ImagePullBackOff", "InvalidImageName"}
-		if status != nil && status.Waiting != nil && containsString(safeReasons, status.Waiting.Reason) {
-			err = r.client.Delete(context.TODO(), &jobs[0], client.PropagationPolicy(metav1.DeletePropagationBackground))
+	// If a job with owner GitOpsConfig exists, but it's blocked because of bad
+	// image, we assume the GitOpsConfig never managed to successfully deploy,
+	// so we can just delete the job, remove the finalizer, and be done (#216).
+	// It may be either action=create or action=delete job.
+	// There can be more than one stuck job - such situation occurs when there
+	// was a GitOpsConfig with a Periodic trigger which created a few jobs. If
+	// there is a job with GitOpsConfig owner, we remove the finalizer just
+	// after removing all stuck jobs - because such job would block a deletion
+	// job creation. Stuck jobs created by a cronJob don't block a deletion job
+	// creation but we delete them nontheless to clean up.
```

Comment:
Can we avoid deleting the jobs created by a CronJob? (Will deleting the CronJob also result in the jobs being automatically deleted?) Would this make our code bigger or smaller (more complex or simpler)? If we could avoid deleting what we don't have to delete in a way that makes the code simpler, it would be awesome. If it increased complexity, I'm ok keeping it as is, I wouldn't want this code to become even more complex than it already is.

If this is equivalent to just moving the loop from l.463-467 higher, I think it might be worth considering. If the block below  just completely ignored CronJob-borne Jobs, it might possibly make the mental context required when reading/analyzing this code somewhat smaller and thus simpler/easier to understand.

Reply:
I updated the code, all stuck jobs have to be deleted